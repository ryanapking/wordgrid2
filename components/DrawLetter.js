import React, { Component } from 'react';
import { Text, StyleSheet } from 'react-native';
import { Container } from 'native-base';
import PropTypes from 'prop-types';

export default class DrawLetter extends Component {
  render() {
    const { letter, letterSize } = this.props;

    if (!letter) {
      return null;
    }

    const textStyles = {
      textAlign: 'center',
      fontSize: letterSize ? letterSize * .75 : 28,
    };

    const borderSize = {
      borderWidth: letterSize ? letterSize * .015 : 1,
    };

    return (
      <Container style={[borderSize, styles.letter, this.props.style]}>
        <Text style={textStyles}>{ letter.toUpperCase() }</Text>
      </Container>
    );
  }

  static propTypes = {
    letter: PropTypes.string,
    letterSize: PropTypes.number,
  }
}

const styles = StyleSheet.create({
  letter: {
    width: "100%",
    height: "100%",
    padding: '5%',
    borderColor: 'white',
    backgroundColor: "#ffd27b",
  },
  full: {
    width: '100%',
    height: '100%'
  }
});