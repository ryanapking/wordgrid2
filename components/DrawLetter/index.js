/* istanbul ignore file */
export { default } from './DrawLetter';