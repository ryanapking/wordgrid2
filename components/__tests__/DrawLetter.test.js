import React from 'react';
import DrawLetter from "../DrawLetter";

import renderer from 'react-test-renderer';

test('DrawLetter renders correctly', () => {
  const drawLetterWithSize = renderer.create(<DrawLetter letter="a" letterSize={25} />).toJSON();
  expect(drawLetterWithSize).toMatchSnapshot();

  const drawLetterWithoutSize = renderer.create(<DrawLetter letter="q" />).toJSON();
  expect (drawLetterWithoutSize).toMatchSnapshot();

  const drawLetterWithoutLetter = renderer.create(<DrawLetter />).toJSON();
  expect (drawLetterWithoutLetter).toMatchSnapshot();
});