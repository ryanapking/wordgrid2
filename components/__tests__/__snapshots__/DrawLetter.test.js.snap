// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderWidth": 0.375,
      },
      Object {
        "backgroundColor": "#ffd27b",
        "borderColor": "white",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "padding": "5%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 18.75,
        "textAlign": "center",
      }
    }
  >
    A
  </Text>
  <Text
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
        },
        Object {
          "fontSize": 6.25,
          "paddingBottom": 0.625,
          "paddingRight": 1.25,
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "borderWidth": 0,
      },
      Object {
        "backgroundColor": "#ffd27b",
        "borderColor": "white",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "padding": "5%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 0,
        "textAlign": "center",
      }
    }
  >
    Q
  </Text>
  <Text
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
        },
        Object {
          "fontSize": 0,
          "paddingBottom": 1,
          "paddingRight": 2,
        },
      ]
    }
  >
    10
  </Text>
</View>
`;

exports[`renders correctly 3`] = `null`;
